Do the VSCode integration steps need to be updated? These work for me in the `main` branch, but not in `ci`. I started from a fresh clone of GoalsARM and switched immediately to the `ci` branch, then opened the folder in VSCode. CMake automatically wrote build files. So far so good.

The first step is to select a kit for GoalsARM, but I do not see that item in the command palette. I don't see "CMake: Select variant" either. If I skip ahead and try to build (Ctrl+Shift+P > CMake: Build) I get the following error:

```
[main] Building folder: C:/Proj/Temp/GoalsARM/build 
[build] Starting build
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --build C:/Proj/Temp/GoalsARM/build --parallel 22 --
[build] MSBuild version 17.5.0+6f08c67f3 for .NET Framework
[build] 
[build] C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppBuild.targets(438,5): error MSB8013: This project doesn't contain the Configuration and Platform combination of Debug|x64. [C:\Proj\Temp\GoalsARM\build\ZERO_CHECK.vcxproj]
[proc] The command: "C:\Program Files\CMake\bin\cmake.EXE" --build C:/Proj/Temp/GoalsARM/build --parallel 22 -- exited with code: 1
[driver] Build completed: 00:00:00.273
[build] Build finished with exit code 1
[rollbar] Unhandled exception: Unhandled Promise rejection: refreshTests Error: Build failed. {}
```

My best guess is that there's at least one additional step needed, but I'm not sure what that would be. In case it's helpful, here is the output I get when I point VSCode at the GoalsARM folder:

```
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" --version
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" -E capabilities
[kit] Successfully loaded 5 kits from C:\Users\RobertGlaubius\AppData\Local\CMakeTools\cmake-tools-kits.json
[presetController] Reading and validating the presets "file C:\Proj\Temp\GoalsARM\CMakePresets.json"
[presetController] Successfully validated C:\Proj\Temp\GoalsARM\CMakePresets.json against presets schema
[presetController] Expanding presets file C:\Proj\Temp\GoalsARM\CMakePresets.json
[presetController] Successfully expanded presets file C:\Proj\Temp\GoalsARM\CMakePresets.json
[main] Configuring project: GoalsARM 
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" "-DCMAKE_CXX_FLAGS=/sdl /guard:cf /utf-8 /diagnostics:caret /w14165 /w44242 /w44254 /w44263 /w34265 /w34287 /w44296 /w44365 /w44388 /w44464 /w14545 /w14546 /w14547 /w14549 /w14555 /w34619 /w34640 /w24826 /w14905 /w14906 /w14928 /w45038 /W4 /permissive- /volatile:iso /Zc:inline /Zc:preprocessor /Zc:enumTypes /Zc:lambda /Zc:__cplusplus /Zc:externConstexpr /Zc:throwingNew /EHsc" "-DCMAKE_EXE_LINKER_FLAGS=/machine:x64 /guard:cf" -DCMAKE_CXX_EXTENSIONS=OFF -DCMAKE_CXX_STANDARD=20 -DCMAKE_CXX_STANDARD_REQUIRED=ON -Dheaderonly_DEVELOPER_MODE=ON -DCMAKE_CONFIGURATION_TYPES=Release -Wdev -Wdeprecated --warn-uninitialized -Werror=dev -Werror=deprecated -SC:/Proj/Temp/GoalsARM -BC:/Proj/Temp/GoalsARM/build -G "Visual Studio 17 2022" -A x64
[cmake] Warn about uninitialized values.
[cmake] -- Selecting Windows SDK version 10.0.22000.0 to target Windows 10.0.22631.
[cmake] -- Locating Catch2
[cmake] -- Could NOT find Catch2 (missing: Catch2_DIR)
[cmake] --   Catch2 not found, fetching dynamically...
[cmake] --   Catch2 fetched succesfully
[cmake] -- Configuring done (1.5s)
[cmake] -- Generating done (0.7s)
[cmake] -- Build files have been written to: C:/Proj/Temp/GoalsARM/build
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -35,6 +35,9 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - uses: seanmiddleditch/gha-setup-ninja@master
+        if: runner.os == 'Linux'
+
       - name: Install boost
         uses: MarkusJx/install-boost@v2
         id: install-boost

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -47,7 +47,7 @@ jobs:
 
       - name: Configure
         shell: pwsh
-        run: cmake "--preset=ci-$("${{ matrix.os }}".split("-")[0])"
+        run: cmake --preset=ci
         env:
           BOOST_ROOT: ${{ steps.install-boost.outputs.BOOST_ROOT }}
 

--- a/.gitignore
+++ b/.gitignore
@@ -3,4 +3,4 @@ cmake-build-*
 .idea
 .vscode
 Testing
-CMakeUserPresets.json
+/CMakeUserPresets.json

--- a/.gitignore
+++ b/.gitignore
@@ -3,3 +3,4 @@ cmake-build-*
 .idea
 .vscode
 Testing
+CMakeUserPresets.json

--- a/CMakePresets.json
+++ b/CMakePresets.json
@@ -1,131 +1,41 @@
 {
-  "version": 6,
-  "cmakeMinimumRequired": {
-    "major": 3,
-    "minor": 15,
-    "patch": 0
-  },
-  "configurePresets": [
-    {
-      "name": "cmake-pedantic",
-      "hidden": true,
-      "warnings": {
-        "dev": true,
-        "deprecated": true,
-        "uninitialized": true,
-        "unusedCli": true,
-        "systemVars": false
-      },
-      "errors": {
-        "dev": true,
-        "deprecated": true
-      }
+    "version": 9,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 25,
+        "patch": 0
     },
-    {
-      "name": "dev-mode",
-      "hidden": true,
-      "inherits": "cmake-pedantic",
-      "cacheVariables": {
-        "headeronly_DEVELOPER_MODE": "ON"
-      }
-    },
-    {
-      "name": "ci-std",
-      "description": "This preset makes sure the project actually builds with at least the specified standard",
-      "hidden": true,
-      "cacheVariables": {
-        "CMAKE_CXX_EXTENSIONS": "OFF",
-        "CMAKE_CXX_STANDARD": "20",
-        "CMAKE_CXX_STANDARD_REQUIRED": "ON"
-      }
-    },
-    {
-      "name": "flags-gcc-clang",
-      "description": "These flags are supported by both GCC and Clang",
-      "hidden": true,
-      "cacheVariables": {
-        "CMAKE_CXX_FLAGS": "-D_GLIBCXX_ASSERTIONS=1 -fstack-protector-strong -fcf-protection=full -fstack-clash-protection -Wall -Wextra -Wpedantic -Wconversion -Wsign-conversion -Wcast-qual -Wformat=2 -Wundef -Werror=float-equal -Wshadow -Wcast-align -Wunused -Wnull-dereference -Wdouble-promotion -Wimplicit-fallthrough -Wextra-semi -Woverloaded-virtual -Wnon-virtual-dtor -Wold-style-cast",
-        "CMAKE_EXE_LINKER_FLAGS": "-Wl,--allow-shlib-undefined,--as-needed,-z,noexecstack,-z,relro,-z,now,-z,nodlopen",
-        "CMAKE_SHARED_LINKER_FLAGS": "-Wl,--allow-shlib-undefined,--as-needed,-z,noexecstack,-z,relro,-z,now,-z,nodlopen"
-      }
-    },
-    {
-      "name": "flags-appleclang",
-      "hidden": true,
-      "cacheVariables": {
-        "CMAKE_CXX_FLAGS": "-fstack-protector-strong -Wall -Wextra -Wpedantic -Wconversion -Wsign-conversion -Wcast-qual -Wformat=2 -Wundef -Werror=float-equal -Wshadow -Wcast-align -Wunused -Wnull-dereference -Wdouble-promotion -Wimplicit-fallthrough -Wextra-semi -Woverloaded-virtual -Wnon-virtual-dtor -Wold-style-cast"
-      }
-    },
-    {
-      "name": "flags-msvc",
-      "description": "Note that all the flags after /W4 are required for MSVC to conform to the language standard",
-      "hidden": true,
-      "cacheVariables": {
-        "CMAKE_CXX_FLAGS": "/sdl /guard:cf /utf-8 /diagnostics:caret /w14165 /w44242 /w44254 /w44263 /w34265 /w34287 /w44296 /w44365 /w44388 /w44464 /w14545 /w14546 /w14547 /w14549 /w14555 /w34619 /w34640 /w24826 /w14905 /w14906 /w14928 /w45038 /W4 /permissive- /volatile:iso /Zc:inline /Zc:preprocessor /Zc:enumTypes /Zc:lambda /Zc:__cplusplus /Zc:externConstexpr /Zc:throwingNew /EHsc",
-        "CMAKE_EXE_LINKER_FLAGS": "/machine:x64 /guard:cf"
-      }
-    },
-    {
-      "name": "ci-linux",
-      "description": "Includes fortification with the CMake default release flags",
-      "inherits": ["flags-gcc-clang", "ci-std"],
-      "generator": "Unix Makefiles",
-      "hidden": true,
-      "cacheVariables": {
-        "CMAKE_BUILD_TYPE": "Release",
-        "CMAKE_CXX_FLAGS_RELEASE": "-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -O3 -DNDEBUG"
-      }
-    },
-    {
-      "name": "ci-darwin",
-      "inherits": ["flags-appleclang", "ci-std"],
-      "generator": "Xcode",
-      "hidden": true
-    },
-    {
-      "name": "ci-win64",
-      "inherits": ["flags-msvc", "ci-std"],
-      "generator": "Visual Studio 17 2022",
-      "architecture": "x64",
-      "hidden": true
-    },
-    {
-      "name": "coverage-linux",
-      "binaryDir": "${sourceDir}/build/coverage",
-      "inherits": "ci-linux",
-      "hidden": true,
-      "cacheVariables": {
-        "ENABLE_COVERAGE": "ON",
-        "CMAKE_BUILD_TYPE": "Coverage",
-        "CMAKE_CXX_FLAGS_COVERAGE": "-Og -g --coverage -fkeep-inline-functions -fkeep-static-functions",
-        "CMAKE_EXE_LINKER_FLAGS_COVERAGE": "--coverage",
-        "CMAKE_SHARED_LINKER_FLAGS_COVERAGE": "--coverage"
-      }
-    },
-    {
-      "name": "ci-build",
-      "binaryDir": "${sourceDir}/build",
-      "hidden": true
-    },
-    {
-      "name": "ci-multi-config",
-      "description": "Speed up multi-config generators by generating only one configuration instead of the defaults",
-      "hidden": true,
-      "cacheVariables": {
-        "CMAKE_CONFIGURATION_TYPES": "Release"
-      }
-    },
-    {
-      "name": "ci-macos",
-      "inherits": ["ci-build", "ci-darwin", "dev-mode", "ci-multi-config"]
-    },
-    {
-      "name": "ci-ubuntu",
-      "inherits": ["ci-build", "ci-linux", "dev-mode"]
-    },
-    {
-      "name": "ci-windows",
-      "inherits": ["ci-build", "ci-win64", "dev-mode", "ci-multi-config"]
-    }
-  ]
+    "include": [
+        "cmake/CMake${hostSystemName}Presets.json"
+    ],
+    "buildPresets": [
+        {
+            "name": "debug",
+            "configurePreset": "default",
+            "configuration": "Debug"
+        },
+        {
+            "name": "release",
+            "configurePreset": "default",
+            "configuration": "Release"
+        }
+    ],
+    "testPresets": [
+        {
+            "name": "debug",
+            "configurePreset": "default",
+            "configuration": "Debug",
+            "output": {
+                "outputOnFailure": true
+            }
+        },
+        {
+            "name": "release",
+            "configurePreset": "default",
+            "configuration": "Release",
+            "output": {
+                "outputOnFailure": true
+            }
+        }
+    ]
 }

--- a/README.md
+++ b/README.md
@@ -12,40 +12,39 @@ HIV transmission dynamic model
 
 ### Build with CMake
 
-First configure the project
+This project uses [CMake](https://cmake.org/) & [CMakePresets](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html) to manage different builds. CMake can manage several stages of dev lifecyle. We will use it to manage configure, build & test. 
+* `configure` generates build system files (Makefiles, Ninja files or VS project files) in the `build` directory, and locates dependencies. It does not compile any code.
+*  `build` this compiles the code and produces outputs (executables, libraries). This will invoke the appropriate build tool (make, ninja or msbuild).
+*  `test` runs tests with `ctest`.
+
+We have the following presets configured for each stage
+* Configure
+    * default
+    * ci
+* build
+    * debug
+    * release
+* test
+    * debug
+    * release
+
+You can use these as follows
+
 ```console
-cmake -S . -B build
+# Configure
+cmake --preset=default
+# Build
+cmake --build --preset=debug
+# Test
+ctest --preset=debug
 ```
-
-This resolves dependencies and generates build system files (Makefiles, Ninja files or VS project files) in the `build` directory. It doesn't compile any code. Then build the project
-
-```console
-cmake --build build
-```
-
-This compiles the code and produces outputs (executables, libraries). This will invoke the appropriate build tool (make, ninja or msbuild).
 
 Note you only need to re-run the configuration step if you modify CMakeLists.txt, or want to set some CMake variable like the build type. If you have just updated source code, just running `cmake --build build` should be enough.
 
-To exclude compiling test code add
-```console
-cmake -S . -B build -DGOALS_BUILD_TESTING=off --fresh
-cmake --build build
-```
-
-Note that if switching between version with tests and without tests, you'll need to reconfigure with the `--fresh` command to ensure that CMake doesn't use cached files.
-
-Note also, when building with testing turned off nothing is going to be compiled, this is a header only library and so this will only fetch dependencies and export a target so this can be compiled downstream.
 
 ### Testing
 
-```console
-cmake -S . -B build
-cmake --build build
-ctest --test-dir build
-```
-
-To add a new test, you can extend the existing `tests/tests.cpp` or any new file within the `tests` directory with the `.cpp` extension will be picked up automatically.
+Testing uses [Catch2](https://github.com/catchorg/Catch2) and is run via ctest. To add a new test, you can extend the existing `tests/tests.cpp` or any new file within the `tests` directory with the `.cpp` extension will be picked up automatically.
 
 ### IDE integration
 
@@ -58,8 +57,10 @@ Should work with CMake out of the box.
 * Install C/C++ extension from Microsoft
 * Install CMake Tools extension from Microsoft
 
-1. Open the project in VSCode, you should be prompted to "Select a Kit for GoalsARM". A kit is the toolchain for compiling your project. You can select a specific kit or "unspecified" to let CMake decide for you.
-1. (Optionally) Select a variant. VSCode will make several build variants avaialble to you, `Debug`, `Release`, `MinRelSize`, `RelWithDebInfo`. `Debug` disables optimisations and includes debug info and should be selected by default. You can select one of the others using Ctrl+Shift+P and running `CMake: Select Variant`. 
-1. You should see VSCode compiling your code automatically. It will create a `build` directory with generated files and compiled code.
+1. At the time of writing, VSCode's CMake extension does not [pick up env variables in preset config](https://github.com/microsoft/vscode-cmake-tools/issues/4253), so we need to work around it. Copy the `cmake/CMakeUserPresets.json` from the `cmake` directory into the root. User presets are a CMake tool for individual users to write specific overrides of configuration in `CMakePresets.json`. Copy the file over, and in the `inherit` block ensure that this is inheriting from the correct presets for your system `cmake/CMakeWindowsPresets.json` on Windows.
+1. Open the project in VSCode.
+1. Specify the configure "preset" by opening command palette (Ctrl+Shift+P) and running "CMake: Select Configure Preset" and select "default".
+1. You should see VSCode compiling your code automatically. It will create a `build` directory with generated files and compiled code. You can turn off automatic compilation and do it manually if you prefer.
+1. (Optionally) Select the build & test presets by using command palette and running "CMake: Select Build Preset" and "CMake: Select Test Preset". Choose either "vscode-debug", to build with debug symbols and no optimisation, or "vscode-release" to build with optimisation and without debug symbols. For most uses you probably want the "vscode-debug" build. It will work without selecting one of these presets, and will build in debug mode.
 1. You should now be able to build and debug by using the "Build" and "Debug" buttons in the bottom bar of VSCode. Note that any compilation buttons shown in the top right won't work as these are not being compiled using the CMake configuration.
 1. You should be able to run individual tests from the testing tab, but to debug from the testing tab you'll need to add a [launch.json](https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/debug-launch.md#debug-using-a-launchjson-file).

--- a/cmake/CMakeCommonPresets.json
+++ b/cmake/CMakeCommonPresets.json
@@ -1,0 +1,56 @@
+{
+    "version": 9,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 30,
+        "patch": 0
+    },
+    "configurePresets": [
+        {
+            "name": "cmake-pedantic",
+            "hidden": true,
+            "warnings": {
+                "dev": true,
+                "deprecated": true,
+                "uninitialized": true,
+                "unusedCli": true,
+                "systemVars": false
+            },
+            "errors": {
+                "dev": true,
+                "deprecated": true
+            }
+        },
+        {
+            "name": "dev-mode",
+            "hidden": true,
+            "inherits": "cmake-pedantic",
+            "cacheVariables": {
+                "GOALS_BUILD_TESTING": "ON"
+            }
+        },
+        {
+            "name": "ci-std",
+            "description": "This preset makes sure the project actually builds with at least the specified standard",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_CXX_EXTENSIONS": "OFF",
+                "CMAKE_CXX_STANDARD": "20",
+                "CMAKE_CXX_STANDARD_REQUIRED": "ON"
+            }
+        },
+        {
+            "name": "ci-build",
+            "binaryDir": "${sourceDir}/build",
+            "hidden": true
+        },
+        {
+            "name": "ci-multi-config",
+            "description": "Speed up multi-config generators by generating only one configuration instead of the defaults",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_CONFIGURATION_TYPES": "Release"
+            }
+        }
+    ]
+}

--- a/cmake/CMakeDarwinPresets.json
+++ b/cmake/CMakeDarwinPresets.json
@@ -1,0 +1,44 @@
+{
+    "version": 9,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 30,
+        "patch": 0
+    },
+    "include": [
+        "CMakeCommonPresets.json"
+    ],
+    "configurePresets": [
+        {
+            "name": "flags-appleclang",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_CXX_FLAGS": "-fstack-protector-strong -Wall -Wextra -Wpedantic -Wconversion -Wsign-conversion -Wcast-qual -Wformat=2 -Wundef -Werror=float-equal -Wshadow -Wcast-align -Wunused -Wnull-dereference -Wdouble-promotion -Wimplicit-fallthrough -Wextra-semi -Woverloaded-virtual -Wnon-virtual-dtor -Wold-style-cast"
+            }
+        },
+        {
+            "name": "ci-darwin",
+            "inherits": [
+                "flags-appleclang",
+                "ci-std"
+            ],
+            "generator": "Xcode",
+            "hidden": true
+        },
+        {
+            "name": "default",
+            "inherits": [
+                "ci-build",
+                "ci-darwin",
+                "dev-mode"
+            ]
+        },
+        {
+            "name": "ci",
+            "inherits": [
+                "default",
+                "ci-multi-config"
+            ]
+        }
+    ]
+}

--- a/cmake/CMakeLinuxPresets.json
+++ b/cmake/CMakeLinuxPresets.json
@@ -1,0 +1,65 @@
+{
+    "version": 9,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 30,
+        "patch": 0
+    },
+    "include": [
+        "CMakeCommonPresets.json"
+    ],
+    "configurePresets": [
+        {
+            "name": "flags-gcc-clang",
+            "description": "These flags are supported by both GCC and Clang",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_CXX_FLAGS": "-D_GLIBCXX_ASSERTIONS=1 -fstack-protector-strong -fcf-protection=full -fstack-clash-protection -Wall -Wextra -Wpedantic -Wconversion -Wsign-conversion -Wcast-qual -Wformat=2 -Wundef -Werror=float-equal -Wshadow -Wcast-align -Wunused -Wnull-dereference -Wdouble-promotion -Wimplicit-fallthrough -Wextra-semi -Woverloaded-virtual -Wnon-virtual-dtor -Wold-style-cast",
+                "CMAKE_EXE_LINKER_FLAGS": "-Wl,--allow-shlib-undefined,--as-needed,-z,noexecstack,-z,relro,-z,now,-z,nodlopen",
+                "CMAKE_SHARED_LINKER_FLAGS": "-Wl,--allow-shlib-undefined,--as-needed,-z,noexecstack,-z,relro,-z,now,-z,nodlopen"
+            }
+        },
+        {
+            "name": "ci-linux",
+            "description": "Includes fortification with the CMake default release flags",
+            "inherits": [
+                "flags-gcc-clang",
+                "ci-std"
+            ],
+            "generator": "Ninja Multi-Config",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Release",
+                "CMAKE_CXX_FLAGS_RELEASE": "-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -O3 -DNDEBUG"
+            }
+        },
+        {
+            "name": "coverage",
+            "binaryDir": "${sourceDir}/build/coverage",
+            "inherits": "ci-linux",
+            "hidden": true,
+            "cacheVariables": {
+                "ENABLE_COVERAGE": "ON",
+                "CMAKE_BUILD_TYPE": "Coverage",
+                "CMAKE_CXX_FLAGS_COVERAGE": "-Og -g --coverage -fkeep-inline-functions -fkeep-static-functions",
+                "CMAKE_EXE_LINKER_FLAGS_COVERAGE": "--coverage",
+                "CMAKE_SHARED_LINKER_FLAGS_COVERAGE": "--coverage"
+            }
+        },
+        {
+            "name": "default",
+            "inherits": [
+                "ci-build",
+                "ci-linux",
+                "dev-mode"
+            ]
+        },
+        {
+            "name": "ci",
+            "inherits": [
+                "default",
+                "ci-multi-config"
+            ]
+        }
+    ]
+}

--- a/cmake/CMakeUserPresets.json
+++ b/cmake/CMakeUserPresets.json
@@ -1,0 +1,41 @@
+{
+    "version": 9,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 25,
+        "patch": 0
+    },
+    "include": [
+        "cmake/CMakeWindowsPresets.json"
+    ],
+    "buildPresets": [
+        {
+            "name": "vscode-debug",
+            "configurePreset": "default",
+            "configuration": "Debug"
+        },
+        {
+            "name": "vscode-release",
+            "configurePreset": "default",
+            "configuration": "Release"
+        }
+    ],
+    "testPresets": [
+        {
+            "name": "vscode-debug",
+            "configurePreset": "default",
+            "configuration": "Debug",
+            "output": {
+                "outputOnFailure": true
+            }
+        },
+        {
+            "name": "vscode-release",
+            "configurePreset": "default",
+            "configuration": "Release",
+            "output": {
+                "outputOnFailure": true
+            }
+        }
+    ]
+}

--- a/cmake/CMakeWindowsPresets.json
+++ b/cmake/CMakeWindowsPresets.json
@@ -1,0 +1,47 @@
+{
+    "version": 9,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 30,
+        "patch": 0
+    },
+    "include": [
+        "CMakeCommonPresets.json"
+    ],
+    "configurePresets": [
+        {
+            "name": "flags-msvc",
+            "description": "Note that all the flags after /W4 are required for MSVC to conform to the language standard",
+            "hidden": true,
+            "cacheVariables": {
+                "CMAKE_CXX_FLAGS": "/sdl /guard:cf /utf-8 /diagnostics:caret /w14165 /w44242 /w44254 /w44263 /w34265 /w34287 /w44296 /w44365 /w44388 /w44464 /w14545 /w14546 /w14547 /w14549 /w14555 /w34619 /w34640 /w24826 /w14905 /w14906 /w14928 /w45038 /W4 /permissive- /volatile:iso /Zc:inline /Zc:preprocessor /Zc:enumTypes /Zc:lambda /Zc:__cplusplus /Zc:externConstexpr /Zc:throwingNew /EHsc",
+                "CMAKE_EXE_LINKER_FLAGS": "/machine:x64 /guard:cf"
+            }
+        },
+        {
+            "name": "ci-win64",
+            "inherits": [
+                "flags-msvc",
+                "ci-std"
+            ],
+            "generator": "Visual Studio 17 2022",
+            "architecture": "x64",
+            "hidden": true
+        },
+        {
+            "name": "default",
+            "inherits": [
+                "ci-build",
+                "ci-win64",
+                "dev-mode"
+            ]
+        },
+        {
+            "name": "ci",
+            "inherits": [
+                "default",
+                "ci-multi-config"
+            ]
+        }
+    ]
+}

--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -31,6 +31,11 @@ foreach(TEST_FILE ${TEST_SOURCES})
     file(RELATIVE_PATH FILE_NAME ${CMAKE_CURRENT_SOURCE_DIR} ${TEST_FILE})
     add_executable(${TEST_NAME} ${FILE_NAME})
     target_include_directories(Catch2 SYSTEM INTERFACE ${catch2_includes})
+    # Suppress warnings for Catch2 when using MSVC
+    if (MSVC)
+        # Per-target suppression (better)
+        set_target_properties(Catch2 PROPERTIES COMPILE_OPTIONS "/w")
+    endif() 
     target_link_libraries(${TEST_NAME} PRIVATE GoalsARM Catch2::Catch2WithMain)
 endforeach()
 

